sanic_ext/extensions/openapi/openapi.py

```diff
@@ -178,24 +178,32 @@ async def handler(request, *handler_args, **handler_kwargs):
 
 def parameter(
     name: Optional[str] = None,
-    schema: Type = str,
-    location: str = "query",
+    schema: Optional[Type] = None,
+    location: Optional[str] = None,
     parameter: Optional[Parameter] = None,
     **kwargs,
```

Comment:
This is feeling to me like a good candidate for [`typing.overload`](https://docs.python.org/3/library/typing.html#typing.overload).

Reply:
> This is feeling to me like a good candidate for [`typing.overload`](https://docs.python.org/3/library/typing.html#typing.overload).

Thanks @ahopkins, that's awesome! I thought Python didn't have method overload, and just learned it has. I will look into it and push an update soon.

Reply:
Overloaded type hints added.